config/opal_check_singularity.m4

```diff
@@ -0,0 +1,52 @@
+# -*- shell-script ; indent-tabs-mode:nil -*-
+#
+# Copyright (c) 2016      Intel, Inc. All rights reserved.
+# $COPYRIGHT$
+#
+# Additional copyrights may follow
+#
+# $HEADER$
+#
+
+# OPAL_CHECK_SINGULARITY(prefix, [action-if-found], [action-if-not-found])
+# --------------------------------------------------------
+AC_DEFUN([OPAL_CHECK_SINGULARITY],[
+    OPAL_VAR_SCOPE_PUSH([spath have_singularity])
+
+    AC_ARG_WITH([singularity],
+                [AC_HELP_STRING([--with-singularity(=DIR)],
+                                [Build support for the Singularity container, optionally adding DIR to the search path (default: no)])],
+                                [], with_singularity=no)
+    spath=
+    AC_MSG_CHECKING([--with-singularity value])
+    AS_IF([test "$with_singularity" = "no"],
+          [AC_MSG_RESULT([no])
+           have_singularity=no],
+          [AS_IF([test -z "$with_singularity" || test "$with_singularity" = "yes"],
+                 [ AC_MSG_RESULT([simple ok (unspecified)])
+                   # look for the singularity command in the default location
+                   AC_CHECK_PROG([SINGULARITY], [singularity], [singularity])
+                   AS_IF([test "$SINGULARITY" != ""],
+                         [have_singularity=yes],
+                         [have_singularity=no])],
```

Comment:
This code path will still allow configure to continue even if a human specifies `--with-singularity` but singularity is not found.